test/performance/load-test/main.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/mako/helpers/go/quickstore"
+	qpb "github.com/google/mako/helpers/proto/quickstore/quickstore_go_proto"
+	vegeta "github.com/tsenart/vegeta/lib"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	deploymentinformer "knative.dev/pkg/injection/informers/kubeinformers/appsv1/deployment"
+	"knative.dev/pkg/signals"
+	netv1alpha1 "knative.dev/serving/pkg/apis/networking/v1alpha1"
+	sksinformer "knative.dev/serving/pkg/client/injection/informers/networking/v1alpha1/serverlessservice"
+
+	"knative.dev/serving/test/performance/mako"
+)
+
+var (
+	benchmark  = flag.String("benchmark", "", "The mako benchmark ID")
+	qps        = flag.Int("qps", 0, "The number of requests to send per second.")
+	masterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+)
+
+func processResults(ctx context.Context, q *quickstore.Quickstore, results <-chan *vegeta.Result) {
+	dl := deploymentinformer.Get(ctx).Lister()
+	sksl := sksinformer.Get(ctx).Lister()
+
+	// Create a ticker to tick every second that prompts us to report a new
+	// summary sample point.
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	// Accumulate the error and request rates along second boundaries.
+	errors := make(map[int64]int64)
+	requests := make(map[int64]int64)
+
+	// When the benchmark completes, iterate over the accumulated rates
+	// and add them as sample points.
+	defer func() {
+		for t, req := range requests {
+			q.AddSamplePoint(mako.XTime(time.Unix(t, 0)), map[string]float64{
+				"rs": float64(req),
+			})
+		}
+		for t, err := range errors {
+			q.AddSamplePoint(mako.XTime(time.Unix(t, 0)), map[string]float64{
+				"es": float64(err),
+			})
+		}
+	}()
+
+	for {
```

Comment:
probably this approach (with a function) would be better in the dataplane probe as well.